openpype/hosts/fusion/plugins/create/create_saver.py

```diff
@@ -125,58 +138,54 @@ def _update_tool_with_data(self, tool, data):
-        # TODO: this should not be done this way - this should actually
-        #       get the data as stored on the tool explicitly (however)
-        #       that would disallow any 'regular saver' to be collected
-        #       unless the instance data is stored on it to begin with
-
-        print("Collecting unmanaged saver..")
-        comp = tool.Comp()
-
-        # Allow regular non-managed savers to also be picked up
-        project = legacy_io.Session["AVALON_PROJECT"]
-        asset = legacy_io.Session["AVALON_ASSET"]
-        task = legacy_io.Session["AVALON_TASK"]
-
-        asset_doc = get_asset_by_name(project_name=project, asset_name=asset)
-
-        path = tool["Clip"][comp.TIME_UNDEFINED]
-        fname = os.path.basename(path)
-        fname, _ext = os.path.splitext(fname)
-        variant = fname.rstrip(".")
-        subset = self.get_subset_name(
-            variant=variant,
-            task_name=task,
-            asset_doc=asset_doc,
-            project_name=project,
+            self._configure_saver_tool(data, tool, subset)
+
+    def _configure_saver_tool(self, data, tool, subset):
+        formatting_data = deepcopy(data)
+
+        # get frame padding from anatomy templates
+        anatomy = Anatomy()
+        frame_padding = int(
+            anatomy.templates["render"].get("frame_padding", 4)
         )
 
+        # Subset change detected
+        workdir = os.path.normpath(legacy_io.Session["AVALON_WORKDIR"])
+        formatting_data.update({
+            "workdir": workdir,
+            "frame": "0" * frame_padding,
+            "ext": "exr"
+        })
+
+        # build file path to render
+        filepath = self.temp_rendering_path_template.format(
+            **formatting_data)
+
+        tool["Clip"] = os.path.normpath(filepath)
+
+        # Rename tool
+        if tool.Name != subset:
+            print(f"Renaming {tool.Name} -> {subset}")
+            tool.SetAttrs({"TOOLS_Name": subset})
+
+    def _collect_saver(self, tool):
+        self.log.info("Collecting saver..")
         attrs = tool.GetAttrs()
+
+        keys = ["id", "asset", "subset", "task", "variant"]
+        ctx_data = {key: tool.GetData(f"openpype.{key}") for key in keys}
         passthrough = attrs["TOOLB_PassThrough"]
         return {
             # Required data
-            "project": project,
-            "asset": asset,
-            "subset": subset,
-            "task": task,
-            "variant": variant,
+            "project": self.project_name,
+            "asset": ctx_data["asset"],
+            "subset": ctx_data["subset"],
+            "task": ctx_data["task"],
+            "variant": ctx_data["variant"],
             "active": not passthrough,
             "family": self.family,
             # Unique identifier for instance and this creator
-            "id": "pyblish.avalon.instance",
+            "id": ctx_data["id"],
             "creator_identifier": self.identifier,
         }
```

Comment:
This function should be removed completely, right? Managed savers are already being collected.

With this logic then a regular saver would start creating errors shown like this:
![image](https://github.com/ynput/OpenPype/assets/2439881/87ff0664-41cf-4c57-8170-711b00e6bba4)

Reply:
Yes, you are right. Removed.